smart-home-provider/cloud/auth-provider.js

```diff
@@ -304,6 +302,52 @@ Auth.registerAuth = function(app) {
 // &grant_type=authorization_code
 
 
+/**
+ * Decode authorization into client_id and client_secret.
+ * @param {Object} req ExpressJS request object
+ * @return {{}}
+ * {
+ *   client_id: "CLIENT_ID",
+ *   client_secret: "CLIENT_SECRET"
+ * }
+ */
+function getHeaderAuthorization(req){
+  try {
+    if (!req.headers.authorization) return {};
+
+    const authorizationString = Buffer.from(req.headers.authorization.split(" ")[1], 'base64').toString();
+    if(!authorizationString) return {};
+
+    const [client_id, client_secret] = authorizationString.split(':');
+    return {
+      client_id,
+      client_secret
+    }
+  } catch (e) {
+    return {};
```

Comment:
This looks great, but can you print this error here just so the developer can see what may have happened?